test/java/org/opendatakit/briefcase/ui/SwingTestRig.java

```diff
@@ -0,0 +1,74 @@
+package org.opendatakit.briefcase.ui;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.CopyOption;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.opendatakit.briefcase.model.BriefcasePreferences;
+import org.opendatakit.briefcase.util.FileSystemUtils;
+
+public class SwingTestRig {
+  private static final Log log = LogFactory.getLog(SwingTestRig.class);
+
+  public static void prepareBriefcaseStorageFolder() {
+    try {
+      BriefcasePreferences.setBriefcaseDirectoryProperty(Files.createTempDirectory("briefcase_test").toString());
+      File briefcaseFolder = new StorageLocation().getBriefcaseFolder();
+      boolean mkdirs = briefcaseFolder.mkdirs();
+      if (mkdirs)
+        log.info("Created test Briefcase storage folder at " + briefcaseFolder);
+      else
+        throw new RuntimeException("Can't create test Briefcase storage folder");
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static void installFormsFrom(Path path) {
+    Path briefcaseStorageFolder = Paths.get(new StorageLocation().getBriefcaseFolder().toURI());
+    uncheckedWalk(path).forEach(sourcePath -> uncheckedCopy(
```

Comment:
I can see why you need uncheckedCopy (because of the stream), but not the other two unchecked..., given that they are each called only once.

![screen shot 2018-01-11 at 1 28 02 pm](https://user-images.githubusercontent.com/55099/34848387-6d23cf28-f6d3-11e7-970c-486f261d87dd.png)

Reply:
it's just to avoid bloating the code with useless try/catch blocks. An alternative would be to declare that this method throws this and that exceptions but my **secret plan** (don't tell anyone) is to use the `Try` monad from Vavr.io to "uncheck" those calls when we add it to the project. Can we agree on leaving it like that in the meantime?